R/getTopTaxa.R

```diff
@@ -0,0 +1,88 @@
+#' Returns top N \dQuote{FeatureID}
+#'
+#' This function is will extract top user specified number of \dQuote{FeatureID}
+#' in a \code{\link[=SummarizedExperiment-class]{SummarizedExperiment}} object.
+#'
+#' @param x A
+#'  \code{\link[=SummarizedExperiment-class]{SummarizedExperiment}} object.
+#'
+#' @param top Numeric value, how many top taxa to return. Default return top five taxa.
+#'
+#' @param method Specify the method to determine top taxa. Either sum, mean or median.
+#'               Default is 'mean'
+#'
+#' @param abund_values a \code{character} value to select an
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assayNames}}
+#'
+#' @return A vector of the top N \dQuote{FeatureID}
+#' @name getTopTaxa
+#'
+#' @author
+#' Sudarshan A. Shetty
+#'
+#' @examples
+#' data(GlobalPatterns, package = "MicrobiomeExperiment")
+#'
+#' top_taxa <- getTopTaxa(GlobalPatterns,
+#'                        method="mean",
+#'                        top=5,
+#'                        abund_values="counts")
+#'
+NULL
+
+#' @rdname getTopTaxa
+#'
+#' @export
+setGeneric("getTopTaxa", signature = "x",
+           function(x, top=5L, method=c("mean","sum","median"), abund_values = "counts")
+               standardGeneric("getTopTaxa"))
+
+
+.check_max_taxa <- function(x, top, abund_values){
+
+    if (!round(top) == top) {
```

Comment:
Nope. this is not a good test. If top is a `character`, what will happen then?

```
> round("a") == "a"
Error in round("a") : non-numeric argument to mathematical function
```

This is the test we are aiming for:
```
if(!is.numeric(top) || as.integer(top) != top){
    top("'top' must be integer value", call. = FALSE)
}
```